app/api/entries/[id]/route.js

```diff
@@ -0,0 +1,40 @@
+import { dbConnect } from '@/utils/dbConnects';
+import Entry from '../models/Entry';
+import { getUserFromCookie } from '@/utils/auth';
+export async function DELETE(req) {
+  await dbConnect();
+  const user = getUserFromCookie();
+  if (!user) {
+    return new Response(JSON.stringify({ error: 'Unauthorized' }), {
```

Comment:
We should probably have a middleware for this
https://nextjs.org/blog/building-apis-with-nextjs#7-building-shared-middleware-logic
As it's not really part of the entries API